packages/cli-core/src/printer/printer.ts

```diff
@@ -88,6 +81,18 @@ export class Printer {
     }
   }
 
+  /**
+   * Logs a message with animated spinner
+   */
+  async indicateProgress(action: () => Promise<void>, message: string) {
+    try {
+      this.startAnimatingSpinner(message);
+      await action();
+    } finally {
+      this.stopAnimatingSpinner(message);
```

Comment:
This is still going to print a check mark even if the action fails. We should probably decouple printing the final message from stopping the spinner. ie `stopAnimatingSpinner` should clear the line and reset the cursor but not print anything.

Then it's on the caller to print a success/fail message. Likely a failure message doesn't need to be printed as the thrown error is already going to be printed. This also solves the problem you had of wanting to print a nice success message like `devDependencies installed!`